From 21487ec89ee9a07e630b2274181a03270134c11c Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:39:14 -0400
Subject: [virt] kvm/vmx: we do rely on loading DR7 on entry

Message-id: <1406040016-3289-160-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86238
O-Subject: [RHEL7 PATCH v2 159/221] KVM: vmx: we do rely on loading DR7 on entry
Bugzilla: 1068627 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Currently, this works even if the bit is not in "min", because the bit is always
set in MSR_IA32_VMX_ENTRY_CTLS.  Mention it for the sake of documentation, and
to avoid surprises if we later switch to MSR_IA32_VMX_TRUE_ENTRY_CTLS.

Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry picked from commit c845f9c646e646e6a5fe416c2e835342984249f7)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kvm/vmx.c b/arch/x86/kvm/vmx.c
index fcc1947..b2a913b 100644
--- a/arch/x86/kvm/vmx.c
+++ b/arch/x86/kvm/vmx.c
@@ -2871,7 +2871,7 @@ static __init int setup_vmcs_config(struct vmcs_config *vmcs_conf)
   !(_vmexit_control & VM_EXIT_ACK_INTR_ON_EXIT))
   _pin_based_exec_control &= ~PIN_BASED_POSTED_INTR;
 
- min = 0;
+ min = VM_ENTRY_LOAD_DEBUG_CONTROLS;
  opt = VM_ENTRY_LOAD_IA32_PAT | VM_ENTRY_LOAD_BNDCFGS;
  if (adjust_vmx_controls(min, opt, MSR_IA32_VMX_ENTRY_CTLS,
     &_vmentry_control) < 0)
-- 
1.7.1